tests/storages_tests/test_journal.py

```diff
@@ -116,3 +128,107 @@ def test_pop_waiting_trial_multiprocess_safe() -> None:
                 if trial_id is not None:
                     trial_id_set.add(trial_id)
         assert len(trial_id_set) == num_enqueued
+
+
+@pytest.mark.parametrize("log_storage_type", LOG_STORAGE_SUPPORTING_SNAPSHOT)
+def test_load_snapshot(log_storage_type: str) -> None:
+    with JournalLogStorageSupplier(log_storage_type) as storage:
+        assert isinstance(storage, optuna.storages.JournalRedisStorage)
+
+        class Loader(object):
+            def __init__(self) -> None:
+                self.n_called = 0
+
+            def __call__(self, snapshot: bytes) -> None:
+                self.n_called += 1
+                raise SnapshotRestoreError
+
+        loader = Loader()
```

Comment:
Can we use `unittest.MagicMock` for this purpose?

Reply:
I'd say we can replace Loader class with MagicMock like https://github.com/HideakiImamura/optuna/compare/feature-journal-redis-snapshot-2...c-bata:optuna:feature-journal-redis-snapshot-2?expand=1